### PR TITLE
Fix timestamp formatting to handle edge cases in milliseconds

### PR DESCRIPTION
**Pull request by bisrat-otterai**
*Description of changes:*

  This PR fixes a bug in the `_format_timestamp` function that could produce invalid timestamps with 4-digit milliseconds. A similar fix was already made to the java plugin: https://github.com/aws/aws-sigv4-auth-cassandra-java-driver-plugin/commit/bfbed565da10739c26c9b321c1fec5c27c5d17bd. Without this fix, there are occasional exceptions that are raised during authentication with a message stating `Unknown SigV4 error`. 

  ## Issue

  When microseconds were 999500 or higher, the rounding logic would produce 1000 milliseconds, resulting in malformed timestamps like `2025-07-06T13:57:58.1000Z` instead of the valid
  `2025-07-06T13:57:58.999Z`.

### CODE REVIEW

File: tests/test_auth.py [hunk 1]
```diff
@@ -104,6 +104,28 @@ def test_generate_scope(self):
...
+        
+    def test_format_timestamp_edge_cases(self):
+        # Test various edge cases
+        test_cases = [
+            (datetime(2025, 1, 1, 0, 0, 0, 0), "2025-01-01T00:00:00.000Z"),
+            (datetime(2025, 1, 1, 0, 0, 0, 500), "2025-01-01T00:00:00.000Z"),  # Truncates to 0ms
+            (datetime(2025, 1, 1, 0, 0, 0, 999000), "2025-01-01T00:00:00.999Z"),
+            (datetime(2025, 1, 1, 0, 0, 0, 999499), "2025-01-01T00:00:00.999Z"),
+            (datetime(2025, 1, 1, 0, 0, 0, 999500), "2025-01-01T00:00:00.999Z"),  # Truncates to 999ms
+        ]
```
Comment (ch-yogya) [inline tests/test_auth.py:123]: Thank you for adding these tests!

**Comment by ch-yogya** [comment]
Thank you for adding detailed CR description.

**Review by ch-yogya — APPROVED** [review]
